src/content/platform-integration/web/embedding-flutter-web.md

```diff
@@ -0,0 +1,326 @@
+pros and cons of `iframe`s.
+
+[`<iframe>`: The Inline Frame element]: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe
+
+## Embedded mode
+
+Flutter web applications can also render content into an arbitrary number of
+elements (commonly `div`s) of another web application; this is called "embedded
+mode" (or "multi-view").
+
+In this mode:
+
+* A Flutter web application can launch, but doesn't render until the first
+"view" is added, with `addView`.
+* The host application may add or remove views from the embedded Flutter web
+application.
+* The Flutter application gets notified of when views are added or removed, so
+it can adjust its widgets accordingly.
+
+### Enable multi-view mode
+
+Enable multi-view mode setting `multiViewEnabled: true` in the
+`initializeEngine` method as shown:
+
+```js highlightLines=8
+// flutter_bootstrap.js
+{% raw %}{{flutter_js}}{% endraw %}
+{% raw %}{{flutter_build_config}}{% endraw %}
+
+_flutter.loader.load({
+  onEntrypointLoaded: async function onEntrypointLoaded(engineInitializer) {
+    let engine = await engineInitializer.initializeEngine({
+      multiViewEnabled: true, // Enables embedded mode.
+    });
+    let app = await engine.runApp();
+    // Make this `app` object available to your JS app.
+  }
+});
+```
+
+### Manage Flutter views from JS
+
+To add or remove views, use the `app` object returned by the `runApp` method:
+
+```js highlightLines=2-4,7
+// Adding a view...
+let viewId = app.addView({
+  hostElement: document.querySelector('#some-element'),
+});
+
+// Removing viewId...
+let viewConfig = flutterApp.removeView(viewId);
+```
+
+### Handling view changes from Dart
+
+View additions and removals are surfaced to Flutter through the
+[`didChangeMetrics` method][] of the `WidgetsBinding` class.
+
+The complete list of views attached to your Flutter app is available
+through the `WidgetsBinding.instance.platformDispatcher.views` iterable. These
+views are of [type `FlutterView`][].
+
+To render content into each `FlutterView`, your Flutter app needs to create a
+[`View` widget][]. `View` widgets can be grouped together under a
+[`ViewCollection` widget][].
+
+The following is an example from the _Multi View Playground_ that encapsulates
+the above in a `MultiViewApp` widget that can be used as the root widget for
+your app. A [`WidgetBuilder` function][] will run for each `FlutterView`:
+
+```dart highlightLines=25,39,46-49,56-61,72
+// multi_view_app.dart
+
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' show FlutterView;
+import 'package:flutter/widgets.dart';
+
+/// Calls [viewBuilder] for every view added to the app to obtain the widget to
+/// render into that view. The current view can be looked up with [View.of].
+class MultiViewApp extends StatefulWidget {
```

Comment:
This seems SUPER complex. It feels like there should be a widget we ship here to do most of this heavy lifting.

Reply:
`MultiViewApp` is (one of the) widgets you're describing; this is how it's used:

* https://github.com/goderbauer/mvp/blob/main/lib/widgets_counter.dart#L9-L11

(The `viewBuilder` function can have additional complexity to decide what widgets go into each `viewId`)